# Sequence-search
Smith-Waterman algorithm, BLOSUM62 matrix, introns/exons

In this project with pretend to implement the Smith-Waterman algorithm (The Smith–Waterman algorithm performs local sequence 
alignment; that is, for determining similar regions between two strings of nucleic acid sequences or protein sequences. Instead of looking at the entire sequence, 
the Smith–Waterman algorithm compares segments of all possible lengths and optimizes the similarity measure.") and use BLOSUM62 
matrix (the BLOSUM (BLOcks SUbstitution Matrix) matrix is a substitution matrix used for sequence alignment of proteins. 
BLOSUM matrices are used to score alignments between evolutionarily divergent protein sequences.") for sequencing alignment.

--------------------------------------------------------------------------------------------------------------------------------
Work made using Python.
Authors: Xavier Pinho and Jorge Melo (@/jorgegomelo)
Course: Introduction to Bioinformatics, University of Coimbra, 2018/2019.
--------------------------------------------------------------------------------------------------------------------------------
